azure-auth-helper/src/main/java/com/microsoft/azure/auth/AzureAuthHelper.java

```diff
@@ -120,6 +202,20 @@ static String baseURL(AzureEnvironment env) {
         return env.activeDirectoryEndpoint() + Constants.COMMON_TENANT;
     }
 
+    private static Object setObjectValue(Object obj, Field field, Object value)
```

Comment:
The method name is a bit confusing for it returns the old value while changes the property

Reply:
how about changeFieldValue?

Reply:
I prefer to create two methods, set and get, for you need to get the origin LogLevel, change the value before device login and set it back.

Reply:
see an example of HashMap.replace

Reply:
Is there an utility function from some existing lib? This function looks fundamental and should have been already implemented somewhere.

Reply:
Updated PR with org.apache.commons.lang3.reflect.FieldUtils